test/src/unittests/rhythm/test_loopbpmestimator.py

```diff
@@ -0,0 +1,120 @@
+        self.assertNotEqual(0, estimate) 
+
+        # Higher than 20% confidence expectation triggers non-zero bpm result in this section 1
+        estimate = LoopBpmEstimator(confidenceThreshold=0.25)(audio[samplePoint1:samplePoint2])                
+        self.assertNotEqual(0, estimate) 
+
+        # Higher than 50% confidence expectation triggers non-zero bpm result in this section 1
+        estimate = LoopBpmEstimator(confidenceThreshold=0.55)(audio[samplePoint1:samplePoint2])                
+        self.assertAlmostEqual(expectedEstimate, estimate, 5) 
+
+        # High confidence expectation triggers 0 bpm result in this section 1
+        estimate = LoopBpmEstimator(confidenceThreshold=0.95)(audio[samplePoint1:samplePoint2])            
+        self.assertEqual(0, estimate) 
+
+        # Randomly chosen points, Section 2
+        # A segment of techno_loop gives a non-ideal estimation of BPM.        
+        samplePoint1 = 22000
+        samplePoint2 = 38000
+        expectedEstimate = 100.0
+
+        # 0% confidence expectation triggers non-zero bpm close to estimate result in this section 2
+        estimate = LoopBpmEstimator(confidenceThreshold=0.0)(audio[samplePoint1:samplePoint2])             
+        self.assertAlmostEqual(expectedEstimate, estimate, 0.1) 
+
+        # 10% confidence expectation triggers non-zero bpm result in this section 2
+        estimate = LoopBpmEstimator(confidenceThreshold=0.1)(audio[samplePoint1:samplePoint2])                
+        self.assertNotEqual(0, estimate) 
+
+        # Higher than 20% confidence expectation triggers 0 bpm result in this section 2
+        estimate = LoopBpmEstimator(confidenceThreshold=0.25)(audio[samplePoint1:samplePoint2])                
+        self.assertEqual(0, estimate) 
+
+        # Higher than 50% confidence expectation triggers 0 bpm result in this section 2
+        estimate = LoopBpmEstimator(confidenceThreshold=0.55)(audio[samplePoint1:samplePoint2])                
+        self.assertEqual(0, estimate) 
+
+        # High confidence expectation triggers 0 bpm result in this section 2
+        estimate = LoopBpmEstimator(confidenceThreshold=0.95)(audio[samplePoint1:samplePoint2])            
+        self.assertEqual(0, estimate) 
+
+    # A series of assert checks on the BPM estimator for empty, zero or constant signals.
+    # LoopBpmEstimator uses the percival estimator internally.
+    # The runtime errors have their origin in that algorithm.
+    def testEmpty(self):
+        emptyAudio = []
+        self.assertRaises(RuntimeError, lambda: LoopBpmEstimator()(emptyAudio))
+
+    def testZero(self):
+        zeroAudio = zeros(1000)
+        self.assertRaises(RuntimeError, lambda: LoopBpmEstimator()(zeroAudio))
+
+    def testConstantInput(self):
+        onesAudio = ones(100)
+        self.assertRaises(RuntimeError, lambda: LoopBpmEstimator()(onesAudio))
```

Comment:
@ffont Is this the desired behavior for LoopBpmEstimator? Specifically, if we were to use this in the Freesound extractor.

Reply:
Maybe we should always give a value on a legit audio input, including silence or constant input. This will be also consistent with RhythmExtractor2013.

Reply:
@ffont, while you are reading this comment, I wanted to comment something on a draft test for percivalbpmestimator

I have it in my local branch ( not yet in a PR).
In this, the audio subsections have been increased to take more significant parts of the loop for estimation.

https://github.com/cvf-bcn-gituser/essentia/blob/unittest_updates/test/src/unittests/rhythm/test_percivalbpmestimator.py

The estimation for full loop is  125.28408813476562, but the estimation for 3 different subsections (50% , 90%) is identical
at 124.90558624267578. This is different from loopbpmestiamtor which returns zero for the 90% section of the same audio.

Reply:
@dbogdanov I think it will be better if it always outputs a value, like `RhythmExtractor2013`. However I'm happy to update the extractor code if needed because it has not yet been deployed to Freesound. We're now working on new analysis system and I'd like to do a full re-analysis after that, using an updated extractor so now it is a good time to make changes if any are needed.

@cvf-bcn-gituser: this might be related to one of my comments above. IIRC loopbpmestiamtor combines the percival bpm estimator with the loop confidence, and outputs 0 if the confidence is below a threshold. The confidence estimator is not prepared for receiving 90% of the loop, it will return 0 (low confidence) if the length is away from a multiple of the estimated BPM beat duration. Does this explain the behaviour you're seeing?

Reply:
@dbogdanov , @ffont is the final consensus now to return a value on a legit audio input, including silence or constant input make it consistent with RhythmExtractor2013?

Reply:
@cvf-bcn-gituser Yes, we all converge towards outputting 0 BPM in such cases, similar to RhythmExtractor2013.

Reply:
Getting loopbpmestimator to converge with RhythmExtractor2013 will require updates in percivalbpmestimator.cpp and this is planned in separate PR.